### ISSUE TITLE
Missing Report with SerenityJunit5Extension

### ISSUE THREAD

**Issue by zeners** [serenity-bdd/serenity-core#2689]
missing generated reports while using junit5 `@ExtendWith`([SerenityJunit5Extension.class](https://github.com/serenity-bdd/serenity-core/blob/master/serenity-junit5/src/main/java/net/serenitybdd/junit5/SerenityJUnit5Extension.java))

to generate reports currently i wrote my own extension, inspired by https://github.com/fabianlinz/serenity-junit5
```
import static net.serenitybdd.core.environment.ConfiguredEnvironment.getConfiguration;
import static net.thucydides.core.steps.StepEventBus.getEventBus;

public class ReportingExtension implements AfterAllCallback {
	@Override
	public void afterAll(ExtensionContext context) {
		final ReportService reportService = new ReportService(getConfiguration());
		final Map<String, TestOutcome> latest = new HashMap<>();
		// report only latest outcome per name
		getEventBus().getBaseStepListener().getTestOutcomes().forEach(o -> latest.put(o.getName(), o));
		reportService.generateReportsFor(List.copyOf(latest.values()));
		reportService.generateConfigurationsReport();
	}
}
```

do i miss something?

**Comment by wakaleo** [comment]
You shouldn't need to do that, the reports should be generated automatically.

**Comment by zeners** [comment]
i assumed that, but they don't show up

i'm using maven-failsfafe-plugin:
```
            <plugin>
                <artifactId>maven-failsafe-plugin</artifactId>
                <configuration>
                    <rerunFailingTestsCount>1</rerunFailingTestsCount>
                    <parallel>classesAndMethods</parallel>
                    <perCoreThreadCount>true</perCoreThreadCount>
                    <threadCount>${threadCount}</threadCount>
                    <!--useUnlimitedThreads>true</useUnlimitedThreads-->
                    <forkCount>${forkCount}</forkCount>
                    <includes>
                        <include>**/*STest.java</include>
                        <include>**/*UITest.java</include>
                    </includes>
                </configuration>
            </plugin>
            <plugin>
                <groupId>net.serenity-bdd.maven.plugins</groupId>
                <artifactId>serenity-maven-plugin</artifactId>
            </plugin>
```

but without my extension neither maven nor intellij run of the tests produce the reports

i don't see what i am missing

**Comment by wakaleo** [comment]
You haven't configured the serenity-maven-plugin to run after test execution - take a look at the starter projects (e.g. https://github.com/serenity-bdd/serenity-junit-starter) to see how to do this.

**Comment by zeners** [comment]
i try (till now with no success) to reproduce the follwing error with the serenity-junit-starter project:
```
[INFO] --- maven-failsafe-plugin:3.0.0-M5:integration-test (default) @ st ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[ERROR] java.util.ServiceConfigurationError: org.junit.platform.launcher.TestExecutionListener: Provider net.serenitybdd.junit5.SerenityTestExecutionListener not found
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 

...

[INFO] --- serenity-maven-plugin:3.1.20:aggregate (serenity-reports) @ st ---
[INFO] Test results for 0 tests generated in 11.5 secs in directory: file:/C:/Users/.../st/target/site/serenity/
[WARNING] No report found on classpath with name single-page.html
[INFO] 
[INFO] --- maven-failsafe-plugin:3.0.0-M5:verify (default) @ st ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:09 min (Wall Clock)
[INFO] Finished at: 2022-02-03T17:01:20+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.0.0-M5:verify (default) on project st: There are test failures.
[ERROR] 
[ERROR] Please refer to C:\Users\...\st\target\failsafe-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[ERROR] java.util.ServiceConfigurationError: org.junit.platform.launcher.TestExecutionListener: Provider net.serenitybdd.junit5.SerenityTestExecutionListener not found
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:733)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:305)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:265)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1314)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1159)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:932)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:190)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:186)
[ERROR] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[ERROR] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[ERROR] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[ERROR] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[ERROR] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
if i use <forkCount>0</forkCount> in <configuration> of maven-failsafe-plugin then it works (without generating Reports) 

I suppose an other component conflicts with the Class-Initializer of SerenityTestExecutionListener
the missing report point to a conflict outside the fork-problem - the error on forking seams only another symptom

looking forward next week

**Comment by zeners** [comment]
finally found:
adding the jacoco-maven-plugin to the linked serenity-junit-starter project giving the error above:

```
            <plugin>
                <groupId>org.jacoco</groupId>
                <artifactId>jacoco-maven-plugin</artifactId>
                <version>0.8.7</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>prepare-agent</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>report</id>
                        <phase>test</phase>
                        <goals>
                            <goal>report</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>

```

**Comment by wakaleo** [comment]
Jacoco is probably trying to instrument the Serenity tests, which will not work, so running Serenity tests with Jacoco is not supported.

**Comment by beyond-danube** [comment]
@wakaleo the issue is not that Jacoco is trying to instrument Serenity tests (we could just exclude those tests from jacoco analysis). The case is that is there are both serenity-junit5 and jacoco, any test run will fail because of dependency itself, even if there are no Serenity JUnit5 tests. 

So the case is not that Serenity tests cannot be instrumented, but that dependency will break execution. While everything works as expected with Serenity JUnit4.

I have created a separate issue for that with sample and clear reproducing scenario #2887

**Comment by wakaleo** [comment]
Could be a dependency conflict between Junit 5 and jacoco - did you do a dependency analysis to look for conflicts?

**Comment by beyond-danube** [comment]
That was my first guess, so I have double-checked that,

There is no conflict between jacoco and junit5 inself (by means of `junit-jupiter-api` and `junit-jupiter-engine`), works OK. Issue appears with `serenity-junit5`, I have checked and it does not have any dependencies intersections.

jacoco:
```
[INFO] Plugin Resolved: jacoco-maven-plugin-0.8.8.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.22.jar
[INFO]     Plugin Dependency Resolved: file-management-1.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-api-3.0.jar
[INFO]     Plugin Dependency Resolved: org.jacoco.agent-0.8.8-runtime.jar
[INFO]     Plugin Dependency Resolved: org.jacoco.core-0.8.8.jar
[INFO]     Plugin Dependency Resolved: org.jacoco.report-0.8.8.jar
```
serenity-junit5
```
[INFO] +- net.serenity-bdd:serenity-junit5:jar:3.2.5:test
[INFO] |  +- org.junit.jupiter:junit-jupiter-params:jar:5.8.2:test
[INFO] |  +- org.junit.platform:junit-platform-launcher:jar:1.8.1:test
[INFO] |  \- org.springframework:spring-core:jar:5.3.16:test
[INFO] |     \- org.springframework:spring-jcl:jar:5.3.16:test
```
